src/components/ResetPassword/ResetPasswordForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useCallback } from 'react';
+import Form from 'react-bootstrap/Form';
+import { Formik } from 'formik';
+import { ButtonType, PrimaryButton } from '../common/Buttons';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { RESET_PASSWORD_VALIDATION_SCHEMA } from './validation';
+import { useResetPasswordMutation } from '../../lib/queries/resetPasswordMutation';
+import { toast } from 'react-toastify';
+import { TextInput } from '../common/Inputs';
+
+type SetPasswordFields = {
+  password: string;
+  repeatPassword: string;
+};
+
+export const ResetPasswordForm: React.FC = () => {
+  const navigate = useNavigate();
+
+  const handleSuccessfulResetPassword = useCallback(() => {
+    navigate('/login');
+  }, [navigate]);
+
+  const { mutate } = useResetPasswordMutation(handleSuccessfulResetPassword);
+
+  const queries = useLocation().search;
+  const params = new URLSearchParams(queries);
+  const token = params.get('token');
+  const userId = params.get('id');
+
+  const handleSubmit = useCallback(
+    async (formFields: SetPasswordFields) => {
+      if (
+        token === undefined ||
+        token === null ||
+        userId === undefined ||
+        userId === null
```

Comment:
Oliko meidän lintteri jotenkin tosi tarkka tästä vaan vai olisko tässä voinu olla vaan


```suggestion
      if (!token || !userId)
```

Reply:
Tää kolmoskerroksen java-ukko tykkäs vaan tollasesta verboosista null-tarkastuksesta :(

Reply:
ESLint: Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly.(@typescript-eslint/strict-boolean-expressions)